modules/ROOT/pages/dataweave-type-reuse-json-schema.adoc

```diff
@@ -0,0 +1,262 @@
+
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from jsonschema!jsonschema!example::schema::Person
+---
+{
+ firstName: "John",
+ age: 20
+} is Root
+----
+
+.Output
+----
+"False"
+----
+
+== Use Definitions Inside Schemas
+
+Definitions inside JSON schemas are also loaded when the module is imported:
+
+.JSON Script
+[source,JSON]
+----
+{
+ "$schema": "http://json-schema.org/draft-04/schema#",
+ "definitions": {
+   "address": {
+     "type": "object",
+     "properties": {
+       "street_address": {
+         "type": "string"
+       },
+       "city": {
+         "type": "string"
+       },
+       "state": {
+         "type": "string"
+       }
+     },
+     "required": [
+       "street_address",
+       "city",
+       "state"
+     ]
+   },
+   "person": {
+     "type": "object",
+     "properties": {
+       "firstName": {
+         "type": "string",
+         "description": "The person's first name."
+       },
+       "lastName": {
+         "type": "string",
+         "description": "The person's last name."
+       },
+       "age": {
+         "description": "Age in years which must be equal to or greater than zero.",
+         "type": "integer",
+         "minimum": 0
+       }
+     },
+     "required": [
+       "firstName",
+       "lastName"
+     ]
+   }
+ },
+ "type": "object",
+ "properties": {
+   "person": {
+     "$ref": "#/definitions/person"
+   },
+   "address": {
+     "$ref": "#/definitions/address"
+   }
+ }
+}
+----
+
+You can import the types from that schema with the following directive:
+
+----
+import * from jsonschema!example::schema::Account
+----
+
+Which has the same effect as declaring these types:
+
+----
+type Root = { address?: address, person?: person }
+
+type address = { street_address: String, city: String, state: String }
+
+type person = { firstName: String, lastName: String, age?: Number }
+----
+
+Use the import directive to import a single type from the schema:
+
+----
+Import address from jsonschema!example::schema::Account
+----
+
+You can change the imported type name to use another name in the script. This helps to avoid the name from clashing with others:
+
+----
+Import address as Account_Address from jsonschema!example::schema::Account
+----
+
+== Facilitate Usage with Pattern Matching
+
+You can use reuse types to take a specific action when an input is of a certain type. These imported types are useful to apply pattern matching.
+
+The following example outputs the value `person` since the object is importing that type definition:
+ 
+.DataWeave Script
+[source,dataweave,linenums]
+----
+%dw 2.0
+import * from import * from jsonschema!example::schema::Account
+---
+{
+ firstName: "John",
+ lastName: "Doe"
+} match {
+case is person -> "PERSON"
+case is address -> "ADDRESS"
+else -> "NO TYPE MATCHED"
+}
+----
+
+.Output
+----
+"Person"
+----
+
+The following example outputs the value `address` since the object is importing that type definition:
```

Comment:
similar question: Is it `address`, `"Address"`, or `"ADDRESS"`? 
and is it importing the `address` type definition from `Account`?

Reply:
same as above.

the type is : address
the string for the pattern matching and output is "ADDRESS"

Reply:
Applying changes as described above.